app/components/subscription_status_component.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class SubscriptionStatusComponent < ViewComponent::Base
```

Comment:
Can you add tests for this file? (If you use the generator `rails g component SubscriptionStatusComponent` this file will automatically be created). Have a look at [the docs for some pointers](https://viewcomponent.org/guide/testing.html), but I'd say:
* One test to check that the component doesn't render if there is no error/warning/empty state (`refute_component_rendered` can help here but isn't mentioned in the testing docs)
* One per status to check that we render the right icon/aria-label